File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnVideoDisc.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* \class GnVideoDisc
* A video disc can be either DVD (Digital Video Disc) or Blu-ray.
*/
public class GnVideoDisc : GnDataObject {
  private HandleRef swigCPtr;

  internal GnVideoDisc(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnVideoDisc obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnVideoDisc() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnVideoDisc(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoDisc_GnType();
    return ret;
  }

  public static GnVideoDisc From(GnDataObject obj) {
    GnVideoDisc ret = new GnVideoDisc(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public GnVideoDisc(string id, string idTag) : this(gnsdk_csharp_marshalPINVOKE.new_GnVideoDisc(id, idTag), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Gracenote ID
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string GnId {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_GnId_get(swigCPtr) );
	} 

  }

/**
* Gracenote unique ID
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string GnUId {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_GnUId_get(swigCPtr) );
	} 

  }

/**
*  Product ID aka Tag ID
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string ProductId {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_ProductId_get(swigCPtr) );
	} 

  }

/**
*  TUI - title-unique identifier
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string Tui {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_Tui_get(swigCPtr) );
	} 

  }

/**
* Tui Tag value
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string TuiTag {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_TuiTag_get(swigCPtr) );
	} 

  }

/**
*  Notes
* @ingroup GDO_ValueKeys_Video
*/
  public string Notes {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoDisc_Notes_get(swigCPtr) );
	} 

  }

/**
*  Ordinal value
* @ingroup GDO_ValueKeys_Misc
*/
  public uint Ordinal {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoDisc_Ordinal_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Matched boolean value indicating whether this type
*   is the one that matched the input criteria.
* @ingroup GDO_ValueKeys_Misc
*/
  public bool Matched {
    get {
      bool ret = gnsdk_csharp_marshalPINVOKE.GnVideoDisc_Matched_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Official title object
* @ingroup GDO_ChildKeys_Title
*/
  public GnTitle OfficialTitle {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoDisc_OfficialTitle_get(swigCPtr);
      GnTitle ret = (cPtr == IntPtr.Zero) ? null : new GnTitle(cPtr, true);
      return ret;
    } 
  }

  public GnVideoSideEnumerable Sides {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoDisc_Sides_get(swigCPtr);
      GnVideoSideEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoSideEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnVideoDisc.java

package com.gracenote.gnsdk;

/** 
** A video disc can be either DVD (Digital Video Disc) or Blu-ray. 
*/ 
 
public class GnVideoDisc extends GnDataObject {
  private long swigCPtr;

  protected GnVideoDisc(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnVideoDisc_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnVideoDisc obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnVideoDisc(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnVideoDisc_gnType();
  }

  public static GnVideoDisc from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnVideoDisc(gnsdk_javaJNI.GnVideoDisc_from(GnDataObject.getCPtr(obj), obj), true);
  }

  public GnVideoDisc(String id, String idTag) throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnVideoDisc(id, idTag), true);
  }

/** 
* Gracenote ID 
* 
*/ 
 
  public String gnId() {
    return gnsdk_javaJNI.GnVideoDisc_gnId(swigCPtr, this);
  }

/** 
* Gracenote unique ID 
* 
*/ 
 
  public String gnUId() {
    return gnsdk_javaJNI.GnVideoDisc_gnUId(swigCPtr, this);
  }

/** 
*  Product ID aka Tag ID 
* 
*/ 
 
  public String productId() {
    return gnsdk_javaJNI.GnVideoDisc_productId(swigCPtr, this);
  }

/** 
*  TUI - title-unique identifier 
* 
*/ 
 
  public String tui() {
    return gnsdk_javaJNI.GnVideoDisc_tui(swigCPtr, this);
  }

/** 
* Tui Tag value 
* 
*/ 
 
  public String tuiTag() {
    return gnsdk_javaJNI.GnVideoDisc_tuiTag(swigCPtr, this);
  }

/** 
*  Ordinal value 
* 
*/ 
 
  public long ordinal() {
    return gnsdk_javaJNI.GnVideoDisc_ordinal(swigCPtr, this);
  }

/** 
*  Matched boolean value indicating whether this type 
*   is the one that matched the input criteria. 
* 
*/ 
 
  public boolean matched() {
    return gnsdk_javaJNI.GnVideoDisc_matched(swigCPtr, this);
  }

/** 
*  Notes 
* 
*/ 
 
  public String notes() {
    return gnsdk_javaJNI.GnVideoDisc_notes(swigCPtr, this);
  }

/** 
*  Official title object 
* 
*/ 
 
  public GnTitle officialTitle() {
    return new GnTitle(gnsdk_javaJNI.GnVideoDisc_officialTitle(swigCPtr, this), true);
  }

  public GnVideoSideIterable sides() {
    return new GnVideoSideIterable(gnsdk_javaJNI.GnVideoDisc_sides(swigCPtr, this), true);
  }

}
